main.queues.tf

```diff
@@ -1,30 +1,32 @@
-resource "azurerm_storage_queue" "this" {
+resource "azapi_resource" "queue" {
   for_each = var.queues
 
-  name                 = each.value.name
-  storage_account_name = azurerm_storage_account.this.name
-  metadata             = each.value.metadata
+  type = "Microsoft.Storage/storageAccounts/queueServices/queues@2023-01-01"
+  body = jsonencode({
```

Comment:
It's recommended to use the dynamic properties which is introduced in the v1.13.1. 

This feature allows user to use HCL object to configure the `body` instead of JSON string.

```
body = {
  properties = {
  }
}
```